### PR TITLE
[NEW] Add Endpoints for Owner, leader, moderator of a group/channel

### PR DESCRIPTION
**Pull request by shubhsherl**
Add Endpoints for Add/Delete Owner, Leader and Moderator of a group/channel.

### CODE REVIEW

**Comment by shubhsherl** [comment]
@filipedelimabrito Can you please review this PR. Required for a PR in RC.Android

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 1]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
         val type = Types.newParameterizedType(RestResult::class.java, DirectMessage::class.java)
 
         return@withContext handleRestCall<RestResult<DirectMessage>>(request, type).result()
-    }
\ No newline at end of file
+    }
+ /* Add owner of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the ` @param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 2]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
...
+        handleRestCall<Any>(request, Any::class.java)
+    }
+}
+
+/**
+ * Add leader of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
+suspend fun RocketChatClient.addLeader(roomId: String ,roomType: RoomType, userId: String) {
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the `@param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 3]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
...
+        handleRestCall<Any>(request, Any::class.java)
+    }
+}
+
+/**
+ * Add Moderator of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
+suspend fun RocketChatClient.addModerator(roomId: String ,roomType: RoomType, userId: String) {
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the `@param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 4]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
...
+        handleRestCall<Any>(request, Any::class.java)
+    }
+}
+
+/**
+ * Remove Owner of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
+suspend fun RocketChatClient.removeOwner(roomId: String ,roomType: RoomType, userId: String) {
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the `@param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 5]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
...
+        handleRestCall<Any>(request, Any::class.java)
+    }
+}
+
+/**
+ * Remove Leader of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
+suspend fun RocketChatClient.removeLeader(roomId: String ,roomType: RoomType, userId: String) {
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the `@param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 6]
```diff
@@ -59,4 +64,143 @@ suspend fun RocketChatClient.createDirectMessage(username: String): DirectMessag
...
+        handleRestCall<Any>(request, Any::class.java)
+    }
+}
+
+/**
+ * Remove Moderator of a Channel/Group
+ * @param roomId Id of the room.
+ * @param roomType The type of the room.
+ */
+suspend fun RocketChatClient.removeModerator(roomId: String ,roomType: RoomType, userId: String) {
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: To be completely documented, it is missing the `@param` for `userId` used here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt [hunk 7]
```diff
@@ -614,4 +615,24 @@ suspend fun RocketChatClient.chatRoomRoles(
         Types.newParameterizedType(List::class.java, ChatRoomRole::class.java)
     )
     return@withContext handleRestCall<RestResult<List<ChatRoomRole>>>(request, type).result()
+}
+
+/**
+ * Ignore User 
+ * @param roomId Id of the room.
+ * @param userId Id of user to remove from room.
+ * @param ignore user.
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt]: I believe we can improve the documentation here.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt [hunk 8]
```diff
@@ -614,4 +615,24 @@ suspend fun RocketChatClient.chatRoomRoles(
...
+ */
+suspend fun RocketChatClient.ignoreUser(roomId: String , userId: String, ignore: Boolean = true): Boolean =
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt]: I think we can move this function to `ChatRoom` class. Actually there is no reason to keep `ChatRoom.kt` and `Channel.kt` on our code. `ChatRoom.kt` is enough.

File: core/src/main/kotlin/chat/rocket/core/internal/model/ChatRoomPayload.kt [hunk 9]
```diff
@@ -9,6 +9,15 @@ data class ChatRoomPayload(@Json(name = "rid") val roomId: String)
...
 data class ChatRoomUnreadPayload(val roomId: String)
 
+@JsonSerializable
+data class ChatRoomIdUserPayload(@Json(name = "rid") val roomId: String, val userId: String)
+
+@JsonSerializable
+data class ChatRoomUserIgnorePayload(@Json(name = "rid") val roomId: String, val userId: String, val ignore: Boolean)
+
+@JsonSerializable
+data class ChatRoomUserPayload(val roomId: String, val userId: String)
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/model/ChatRoomPayload.kt]: Isn't the json name for the `roomId` parameter `rid`?

File: core/src/main/kotlin/chat/rocket/core/internal/model/ChatRoomPayload.kt [hunk 10]
```diff
@@ -9,6 +9,15 @@ data class ChatRoomPayload(@Json(name = "rid") val roomId: String)
...
+
 @JsonSerializable
 data class ChatRoomNamePayload(val roomId: String, val name: String?)
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/model/ChatRoomPayload.kt]: Isn't the json name for the `roomId` parameter `rid`?

**Pull request merged**